include/aws/io/event_loop.h

```diff
@@ -116,6 +117,15 @@ AWS_EXTERN_C_BEGIN
 AWS_IO_API
 void aws_event_loop_schedule_task_now(struct aws_event_loop *event_loop, struct aws_task *task);
 
+/**
+ * Variant of aws_event_loop_schedule_task_now that forces all tasks to go through the cross thread task queue,
+ * guaranteeing that order-of-submission is order-of-execution.  If you need this guarantee, you must use this
+ * function; the base function contains short-circuiting logic that breaks ordering invariants.  Beyond that, all
+ * properties of aws_event_loop_schedule_task_now apply to this function as well.
+ */
+AWS_IO_API
+void aws_event_loop_schedule_task_now_serialized(struct aws_event_loop *event_loop, struct aws_task *task);
```

Comment:
> guaranteeing that order-of-submission is order-of-execution

But if a task is cancelled, that task will not be following this order. Let's point this out in case we forget it in the future.

Reply:
Using cancel not only breaks order, it also can happen on any thread.  There's not much we can do there.